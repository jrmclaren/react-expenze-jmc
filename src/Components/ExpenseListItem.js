import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ( {description, amount, createdAt, id} = {} ) => (
			<div>
				<hr />
				<Link to={`/edit/${id}`}>
					<h3>{description}</h3>
				</Link>
				<p>Amount: {amount}</p>
				<p> Timestamp: {createdAt}</p>

			</div> 
		);
export default ExpenseListItem;
