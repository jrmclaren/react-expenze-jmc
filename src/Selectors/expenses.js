import moment from 'moment';
// Get visible expenses
export default (expenses, {text, sortBy, startDate, endDate} ) => {
	return expenses.filter( ( expense ) => {
			const createdAtMoment = moment(expense.createdAt)
			let startDateMatch =  startDate? startDate.isSameOrBefore(createdAtMoment, 'day') : true;
			let endDateMatch = endDate? endDate.isSameOrAfter(createdAtMoment, 'day') : true;
			let textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
			// expenses.description as the text variable string inside of it
			// includes(); if one string includes another
			// convert both strings to lower case
			// 
			return startDateMatch && endDateMatch && textMatch;
	}).sort(( a, b  ) => {
		if( sortBy === 'date' ){
			return (a.createdAt < b.createdAt? 1 : -1);
		} else if( sortBy === 'amount') {
			return a.amount < b.amount? 1 : -1;
		}
	});
}
