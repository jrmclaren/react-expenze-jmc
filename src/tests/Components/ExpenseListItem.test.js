import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../Components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render expense list item using fixtures expense [0]: ', () => {
  const expense = expenses[0];
  const wrapper = shallow(<ExpenseListItem {...expense} />);
  expect(wrapper).toMatchSnapshot();
});
