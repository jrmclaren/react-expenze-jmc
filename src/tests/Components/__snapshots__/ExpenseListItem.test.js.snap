// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item using fixtures expense [0]:  1`] = `
<div>
  <hr />
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      gum
    </h3>
  </Link>
  <p>
    Amount: 
    195
  </p>
  <p>
     Timestamp: 
    0
  </p>
</div>
`;
